Ubuntu 19.10 Installation

Hello, and thank you for starting this wonderful project.

I'm trying to compile the examples so that I can run the live demo as I want to view the remarkable screen on my computer screen.

I used this comment to install the dependencies https://github.com/canselcik/libremarkable/issues/5#issuecomment-377592830. There were no errors and I could follow all steps.

However, when I run the last step I get the following error:

```
$ make
cargo build --release --target=armv7-unknown-linux-gnueabihf
    Finished release [optimized + debuginfo] target(s) in 0.10s
cargo build --examples --release --target=armv7-unknown-linux-gnueabihf
   Compiling libremarkable v0.3.2 (/home/viktsend/Programs/libremarkable)
warning: use of deprecated item 'std::sync::ONCE_INIT': the `new` function is now preferred
   --> examples/spy.rs:106:1
    |
106 | / hook! {
107 | |   unsafe fn ioctl(fd: c_int, request: NativeWidthType, p1: intptr_t, p2: intptr_t, p3: intptr_t, p4: intptr_t) -> c_int => ioctl_hook {
108 | |     if request == FBIOPUT_VSCREENINFO {
109 | |         let info = p1 as *mut VarScreeninfo;
...   |
142 | |   }
143 | | }
    | |_^
    |
    = note: `#[warn(deprecated)]` on by default
    = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

warning: use of deprecated item 'std::sync::ONCE_INIT': the `new` function is now preferred
   --> examples/spy.rs:106:1
    |
106 | / hook! {
107 | |   unsafe fn ioctl(fd: c_int, request: NativeWidthType, p1: intptr_t, p2: intptr_t, p3: intptr_t, p4: intptr_t) -> c_int => ioctl_hook {
108 | |     if request == FBIOPUT_VSCREENINFO {
109 | |         let info = p1 as *mut VarScreeninfo;
...   |
142 | |   }
143 | | }
    | |_^
    |
    = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error: lto can only be run for executables, cdylibs and static library outputs

error: aborting due to previous error; 2 warnings emitted

error: could not compile `libremarkable`.

To learn more, run the command again with --verbose.
make: *** [Makefile:7: examples] Error 101

17:18 viktsend@stealth
~/Programs/libremarkable
$ 
```

I'm running Ubuntu 19.10. Any help will be much appreciated.

Best
V

Demo didn't run, unfortunately:

```
$ make run
cargo build --examples --release --target=armv7-unknown-linux-gnueabihf
warning: unused manifest key: build
warning: unused manifest key: package.is-it-maintained-issue-resolution
warning: unused manifest key: package.is-it-maintained-open-issues
warning: unused manifest key: package.maintenance
warning: unnecessary braces around `for` iterator expression
  --> src/framebuffer/draw.rs:84:18
   |
84 |         for y in { (-search_distance)..search_distance } {
   |                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: remove these braces
   |
   = note: `#[warn(unused_braces)]` on by default

warning: 1 warning emitted

warning: use of deprecated item 'std::sync::ONCE_INIT': the `new` function is now preferred
   --> examples/spy.rs:106:1
    |
106 | / hook! {
107 | |   unsafe fn ioctl(fd: c_int, request: NativeWidthType, p1: intptr_t, p2: intptr_t, p3: intptr_t, p4: intptr_t) -> c_int => ioctl_hook {
108 | |     if request == FBIOPUT_VSCREENINFO {
109 | |         let info = p1 as *mut VarScreeninfo;
...   |
142 | |   }
143 | | }
    | |_^
    |
    = note: `#[warn(deprecated)]` on by default
    = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

warning: use of deprecated item 'std::sync::ONCE_INIT': the `new` function is now preferred
   --> examples/spy.rs:106:1
    |
106 | / hook! {
107 | |   unsafe fn ioctl(fd: c_int, request: NativeWidthType, p1: intptr_t, p2: intptr_t, p3: intptr_t, p4: intptr_t) -> c_int => ioctl_hook {
108 | |     if request == FBIOPUT_VSCREENINFO {
109 | |         let info = p1 as *mut VarScreeninfo;
...   |
142 | |   }
143 | | }
    | |_^
    |
    = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

warning: 2 warnings emitted

    Finished release [optimized + debuginfo] target(s) in 0.11s
ssh root@"10.11.99.1" 'kill -9 `pidof demo` || true; systemctl stop xochitl || true'
root@10.11.99.1's password: 
Permission denied, please try again.
root@10.11.99.1's password: 
kill: usage: kill [-s sigspec | -n signum | -sigspec] pid | jobspec ... or kill -l [sigspec]
scp ./target/armv7-unknown-linux-gnueabihf/release/examples/demo root@"10.11.99.1":~/
root@10.11.99.1's password: 
demo                                                                                                                                                                               100%   46MB   6.3MB/s   00:07    
ssh root@"10.11.99.1" './demo'
root@10.11.99.1's password: 
./demo: /lib/libm.so.6: version `GLIBC_2.29' not found (required by ./demo)
./demo: /lib/libc.so.6: version `GLIBC_2.28' not found (required by ./demo)
make: *** [Makefile:41: deploy-demo] Error 1

10:38 viktsend@stealth
~/Programs/libremarkable
$ make start-xochitl
ssh root@"10.11.99.1" 'kill -9 `pidof demo` || true; systemctl start xochitl'
root@10.11.99.1's password: 
kill: usage: kill [-s sigspec | -n signum | -sigspec] pid | jobspec ... or kill -l [sigspec]


```

--- a/README.md
+++ b/README.md
@@ -19,25 +19,25 @@ For further documentation see the [wiki](https://github.com/canselcik/libremarka
 ### Build Instructions
 
 #### Setting up the toolchain
-In order to build `libremarkable` and the examples (`spy.so` and `demo`), you'll need the following configuration after having installed the proper toolchain to your `$PATH`. The `arm-linux-gnueabihf-gcc` toolchain is used to build both implementations.
-
-The toolchain that would be acquired from either of these sources would be able to cross-compile for the Remarkable Tablet:
-```
-AUR:
-  https://aur.archlinux.org/packages/arm-linux-gnueabihf-gcc/
-Remarkable:
-  https://remarkable.engineering/deploy/sdk/poky-glibc-x86_64-meta-toolchain-qt5-cortexa9hf-neon-toolchain-2.1.3.sh
-```
+In order to build `libremarkable` and the examples (`spy.so` and `demo`), you'll need the toolchain from Remarkable. Download the installation script from https://remarkable.engineering/deploy/sdk/poky-glibc-x86_64-meta-toolchain-qt5-cortexa9hf-neon-toolchain-2.1.3.sh and install the toolchain.
 
 You can then set up your Rust toolchain for cross compilation with: `rustup target add armv7-unknown-linux-gnueabihf`.
 
-Once that's done, you should add the following to your `~/.cargo/config`:
+Once that's done, you should add the following to `.cargo/config` (replace `<path-to-installed-poky-toochain>` with the directory you installed the Remarkable toolchain to):
 ```
 [target.armv7-unknown-linux-gnueabihf]
-linker = "arm-linux-gnueabihf-gcc"
+linker = "<path-to-the-installed-poky-toolchain>/sysroots/x86_64-pokysdk-linux/usr/bin/arm-poky-linux-gnueabi/arm-poky-linux-gnueabi-gcc"
+rustflags = [
+  "-C", "link-arg=-march=armv7-a",
+  "-C", "link-arg=-marm",
+  "-C", "link-arg=-mfpu=neon",
+  "-C", "link-arg=-mfloat-abi=hard",
+  "-C", "link-arg=-mcpu=cortex-a9",
+  "-C", "link-arg=--sysroot=<path-to-the-installed-poky-toolchain>/sysroots/cortexa9hf-neon-poky-linux-gnueabi",
+]
 ```
 
-If you have further questions, feel free to ask in Issues or take a look at [this comment](https://github.com/canselcik/libremarkable/issues/5#issuecomment-377592830) on one of the issues under this repo.
+If you have further questions, feel free to ask in Issues.
 
 #### Building libremarkable and the examples
 A simple Makefile wrapper is created for convenience. It exposes the following verbs:

